### PR TITLE
feat: implement message collection hooks

### PR DESCRIPTION
**Pull request by bang9**

### CODE REVIEW

**Comment by xtaraim** [comment]
hi here me also

File: src/lib/Sendbird.tsx [hunk 1]
```diff
@@ -375,10 +375,11 @@ const SendbirdSDK = ({
             sdkInitialized && configsWithAppAttr(sdk).groupChannel.setting.enableMessageSearch,
           // Remote configs set from dashboard by UIKit feature configuration
           groupChannel: {
-            enableOgtag:
-              sdkInitialized && configsWithAppAttr(sdk).groupChannel.channel.enableOgtag,
+            enableOgtag: sdkInitialized && configsWithAppAttr(sdk).groupChannel.channel.enableOgtag,
             enableTypingIndicator: configs.groupChannel.channel.enableTypingIndicator,
             enableDocument: configs.groupChannel.channel.input.enableDocument,
+            enableReactions: configs.groupChannel.channel.enableReactions,
+            replyType: configs.groupChannel.channel.replyType,
```
Comment (HoonBaek) [inline src/lib/Sendbird.tsx:382]: 👍

File: src/lib/emojiManager.tsx [hunk 2]
```diff
@@ -60,4 +60,8 @@ export class EmojiManager {
   public getEmojiUrl(reactionKey: Reaction['key']) {
     return this.AllEmojisAsArray.find((emoji) => emoji.key === reactionKey).url ?? '';
   }
+
+  public get emojiContainer() {
```
Comment (HoonBaek) [inline src/lib/emojiManager.tsx:64]: 🤩👍

File: src/modules/GroupChannel/components/MessageList/getMessagePartsInfo.ts [hunk 3]
```diff
@@ -39,9 +39,9 @@ export const getMessagePartsInfo = ({
   const previousMessageCreatedAt = previousMessage?.createdAt;
   const currentCreatedAt = currentMessage.createdAt;
   // https://stackoverflow.com/a/41855608
-  const hasSeparator = !(previousMessageCreatedAt && (
-    isSameDay(currentCreatedAt, previousMessageCreatedAt)
-  ));
+  const hasSeparator = 'sendingStatus' in currentMessage && currentMessage.sendingStatus !== 'succeeded' // NOTE: for pending/failed messages
+    ? false
+    : !(previousMessageCreatedAt && (isSameDay(currentCreatedAt, previousMessageCreatedAt)));
```
Comment (HoonBaek) [inline src/modules/GroupChannel/components/MessageList/getMessagePartsInfo.ts:44]: Can we wrap on the ternary operator with parentheses? for readability
```suggestion
  const hasSeparator = 'sendingStatus' in currentMessage && (
    currentMessage.sendingStatus !== 'succeeded'
      ? false
      : !(previousMessageCreatedAt && (isSameDay(currentCreatedAt, previousMessageCreatedAt)))
      );
```

**Review by HoonBaek — APPROVED** [review]
LGTM!

File: src/modules/GroupChannel/components/MessageList/getMessagePartsInfo.ts (hunk 3 above)
Comment (bang9) [reply src/modules/GroupChannel/components/MessageList/getMessagePartsInfo.ts:44]: sure~

**Pull request merged**